### PR TITLE
46 update user views

### LINKED ISSUE #46: [Users] Frontend-team Updates to User Model, Views
Discussions with frontend team require a few things to be changed around the AppUser/User:

- The AppUser model must now support a new field: an `is_form_submitted` flag, that is set once a specific professor submits their teaching preferences form for the current year.
- the /users views must be updated to support a new endpoint available to both logged-in Admins and Profs: `GET api/user`, which gets the information for the user logged in (no password!).
- Additionally, a double-check should be done that the Admin endpoint `GET api/users` should only return Professor accounts (non-admin)
- Update it so when we delete a Prof Account (AppUser), it also deletes the corresponding User Object.
-----
Extra update that is non-essential to this sprint:
- The FE team wants Preferences record to display default (blank) values when a Prof has not submitted their Preferences form. We should update the Prof account creation logic to also create an empty associated Preferences record for that prof.

### PR DESCRIPTION
**Pull request by skosman**
Updated users to reflect changes discussed with FE team:
* added is_form_submitted field to appUser
* removed password from response
* created /user endpoint to get logged-in user information
* added to get prof list test that admins are not included (we use a filter in the appUsers query to only get non-super users)
* added signal to delete user when appuser is deleted

Will need to update the tests once Russ' PR is in. Will also create another ticket to update prof account creation logic.

/User Endpoint
image

Closes #46

### CODE REVIEW

File: users/models.py [hunk 1]
```diff
@@ -32,11 +32,17 @@ class TeachingType(models.TextChoices):
...
     is_peng = models.BooleanField(default=False)
+    is_form_submitted = models.BooleanField(default=False)
 
     class Meta:
         managed = True  #auto creates tables
         db_table = 'appuser'
 
+# Use Django signals to delete User instance when AppUser is deleted. Based on: https://stackoverflow.com/a/12754229
+@receiver(post_delete, sender=AppUser)
+def delete_code_constraint_with_question(sender, instance, **kwargs):
```
Comment (rbassot) [inline users/models.py:43]: Is this a desired class method name? I think it is just a bit confusing to me

Comment (skosman) [reply users/models.py:43]: Nice catch! I'll fix that

Comment (skosman) [reply users/models.py]: I updated the name

**Comment by rbassot** [comment]
When logged in as a admin superuser account, the `GET api/user` endpoint responds with a 404 with no JSON body. This could be a due to a consequence of filtering out admin accounts for other requests, but this endpoint should be able to support admin-side accounts, according to FE team:

image

image

**Comment by skosman** [comment]
> When logged in as a admin superuser account, the `GET api/user` endpoint responds with a 404 with no JSON body. This could be a due to a consequence of filtering out admin accounts for other requests, but this endpoint should be able to support admin-side accounts, according to FE team:
> 
> image
> 
> image

Is your admin have an associated app user? I have it set so it queries from all app users which I thought also included admins.

**Comment by rbassot** [comment]
> 

You're completely right, my AppUser got deleted on the way somewhere. Works now! This is something that we should watch out for when the working system admin users are going to be created - creating Django admin accounts won't create associated AppUser instances

File: users/models.py (hunk 1 above)
Comment (rbassot) [reply users/models.py]: Looks good thanks!

**Comment by skosman** [comment]
> > 
> 
> You're completely right, my AppUser got deleted on the way somewhere. Works now! This is something that we should watch out for when the working system admin users are going to be created - creating Django admin accounts won't create associated AppUser instances

I ran into the same issue that's why I know haha we should definitely watch out for it. I can update the read to say create an associated appUser when creating an admin

**Comment by rbassot** [comment]
- new AppUser field works as expected, viewable in Admin page
- AppUser signal to delete associated User works as expected (Admin page & through Postman request)
- `password` field no longer appears in either GET /users or GET /user/ for both admin & prof-side accounts
- test suites pass and look good

Great work and thank you for all this!

**Review by rbassot — APPROVED** [review]

**Pull request merged**